src/invariant/OrderBookIsNotCrossed.h

```diff
@@ -30,6 +30,14 @@ struct OfferEntryCmp
         {
             return true;
         }
+        else if (((a.flags & PASSIVE_FLAG) == 0) &&
```

Comment:
This makes `OfferEntryCmp` violate the asymmetry requirement of https://en.cppreference.com/w/cpp/named_req/Compare.

Let `x < y` denote `OfferEntryCmp()(x, y)` for any `OfferEntry x, y`. If
```
L: Price = 1/1, Flags = PASSIVE_FLAG (offerID doesn't matter in this example)
R: Price = 2/1, Flags = 0 (offerID doesn't matter in this example)
```
Then

- `L < R == true` because `price(L) < price(R) == true`
- `R < L == true` because `price(R) < price(L) == false` and `(R.flags & PASSIVE_FLAG == 0) && (L.flags & PASSIVE_FLAG != 0) == true`

Reply:
Thanks -- that test is supposed to be inside the prices-equal case, of course.  Sigh.  Fixed.